Documentation/ceph-csi-troubleshooting.md

```diff
@@ -0,0 +1,395 @@
+---
+title: CSI Common Issues
+weight: 11125
+indent: true
+---
+
+# CSI Common Issues
+
+Issues when provisioning volumes with the Ceph CSI driver can happen for many reasons such as:
+
+- Network connectivity between CSI pods and ceph
+- Cluster health issues
+- Slow operations
+- Kubernetes issues
+- Ceph-CSI configuration or bugs
+
+The following troubleshooting steps can help identify a number of issues.
+
+### Block (RBD)
+
+If you are mounting block volumes (usually RWO), these are referred to as `RBD` volumes in Ceph.
+See the sections below for RBD if you are having block volume issues.
+
+### Shared Filesystem (CephFS)
+
+If you are mounting shared filesystem volumes (usually RWX), these are referred to as `CephFS` volumes in Ceph.
+See the sections below for CephFS if you are having filesystem volume issues.
+
+## Network Connectivity
+
+The Ceph monitors are the most critical component of the cluster to check first.
+Retrieve the mon endpoints from the services:
+
+```console
+$ kubectl -n rook-ceph get svc -l app=rook-ceph-mon
```

Comment:
If we believe the output is important, then I'd do 2 separate blocks, one where we can easily copy the command (since it's generic and not specific to our example) without getting the `$`  and one with the output.